src/Randomizer.SMZ3.Tracking/Configuration/ConfigTypes/ItemData.cs

```diff
@@ -318,15 +318,47 @@ public bool TryGetTrackingResponse([NotNullWhen(true)] out SchrodingersString? r
         /// cref="Tracker.IsWorth(ItemData)"/> to include items that this item
         /// might logically lead to.
         /// </remarks>
-        public bool IsJunk(Config config)
+        public bool IsJunk(Config? config)
         {
-            var junkCategories = config.Keysanity
-                ? new[] { ItemCategory.Junk, ItemCategory.Scam, ItemCategory.Map, ItemCategory.Compass }
-                : new[] { ItemCategory.Junk, ItemCategory.Scam, ItemCategory.Map, ItemCategory.Compass,
-                    ItemCategory.SmallKey, ItemCategory.BigKey, ItemCategory.Keycard };
+            if (InternalItemType == ItemType.Nothing || InternalItemType.IsInAnyCategory(new[] { ItemCategory.Junk, ItemCategory.Scam, ItemCategory.NonRandomized, ItemCategory.Compass }))
+                return true;
+
+            if (config?.ZeldaKeysanity == true && InternalItemType.IsInAnyCategory(new[] { ItemCategory.SmallKey, ItemCategory.BigKey, ItemCategory.Map }))
+                return true;
 
-            return InternalItemType == ItemType.Nothing
-                || InternalItemType.IsInAnyCategory(junkCategories);
+            if (config?.MetroidKeysanity == true && InternalItemType.IsInCategory(ItemCategory.Keycard))
+                return true;
+
+            return false;
+        }
+
+        /// <summary>
+        /// Determines whether the item is progression given the specified
+        /// configuration, assuming one is provided.
+        /// </summary>
+        /// <param name="config">The randomizer configuration.</param>
+        /// <returns>
+        /// <c>true</c> if the item is considered progression; otherwise, <c>false</c>.
+        /// </returns>
+        /// <remarks>
+        /// This method only considers the item's value on its own. Call <see
+        /// cref="Tracker.IsWorth(ItemData)"/> to include items that this item
+        /// might logically lead to.
+        /// </remarks>
+        public bool IsProgression(Config? config)
+        {
+            if (InternalItemType == ItemType.Nothing || InternalItemType.IsInAnyCategory(new[] { ItemCategory.Junk, ItemCategory.Scam, ItemCategory.NonRandomized, ItemCategory.Map, ItemCategory.Compass, ItemCategory.Nice }))
+                return false;
+
+            if (InternalItemType.IsInAnyCategory(new[] { ItemCategory.SmallKey, ItemCategory.BigKey }))
+                return config?.ZeldaKeysanity == true;
+
+            if (InternalItemType.IsInCategory(ItemCategory.Keycard))
+                return config?.MetroidKeysanity == true;
+
+            // Todo: We can add special logic like checking if it's one of the first two swords
```

Comment:
If we want to make the "item worth" checks any more complex we should probably move it into ItemService or something so we can do more than simple static checking.

Reply:
Yeah, that's fair. I think there are also a few other places that have similar logic that would be good to consolidate as well. I'll make a note in the issue for that.